File: spark/src/main/java/com/ithiama/spark/WordCount.java
package com.ithiama.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;


import java.util.Arrays;

/**
 * @program: spark
 * @author: Mr.Jiang
 * @create: 2019-08-24 14:03
 * @description:
 **/
public class WordCount {

    public static void main(String[] args) {

        //todo 1.构建spark程序application的上下文对象
        SparkConf config = new SparkConf().setAppName("wordCount").setMaster("local[2]");
        JavaSparkContext sc = new JavaSparkContext(config);

        //todo 2.读取文件
        JavaRDD<String> inputRDD = sc.textFile("/datas/sparkInput/wc.txt");

        //todo 3.切割压缩
        JavaPairRDD<String, Integer> wordCountsRDD = inputRDD.flatMap(line -> Arrays.asList(line.trim().split(" ")).iterator())
                .mapToPair(word -> new Tuple2<>(word, 1))
                .reduceByKey(((v1, v2) -> v1 + v2));

        wordCountsRDD.foreach(tuple-> System.out.println("tuple = " + tuple.toString()));

        sc.stop();
    }
}

File: spark_2.11/spark_day02/src/main/scala/com/itheima/spark/iter/WordCount.scala
package com.itheima.spark.iter

import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}

object WordCount {

  def main(args: Array[String]): Unit = {

    //todo 1.构建spark程序的入口SparkContext上下文实例对象
    val sparkConf: SparkConf = new SparkConf()
      .setAppName("SparkWordCount")
      .setMaster("local[1]") //以本地模式运行

    //参数:config: SparkConf  需要SparkConf对象
    val sc: SparkContext = new SparkContext(sparkConf)


    val inputRDD: RDD[String] = sc.textFile("./datas/group/group.data")

    val wordCountRDD: RDD[(String, String)] = inputRDD.mapPartitions { iter =>
      iter.map(line => (line.split("\\s+")(0), line.split("\\s+")(1))) }

    wordCountRDD.foreachPartition { iter =>
      iter.foreach { case (k, v) => println(s"$k->$v") }
    }
  }
}